### ISSUE TITLE
RosettaNet wait for Response Action failed

### ISSUE THREAD

**Issue by Prafulkapse** [Azure/logicapps#630]
RosettaNet wait for response does not generate a Notification of failure message after exhausting retries. It gives below error 
"InternalServerError. Encountered internal server error. The tracking Id is 'e289fc38-2be3-4897-8627-d8bbf4df5d98'."
The error is not at all helpful. 
Kindly provide a good detailed error to know where exactly is the problem

**Comment by praveensri** [comment]
Hi @Prafulkapse , Can you please provide more details on recent run id, workflow name and region? The tracking id provided seems to be old ones which we do not have much details now.

**Comment by Prafulkapse** [comment]
Hi @praveensri:

Please find the details below.

Logic App: LA-GNH-RN-FLEX-OUTBOUND-WEU-TST
Run ID: 08585319578026371913289292374CU32
Region : West Europe

BR,
Praful Kapse

**Comment by praveensri** [comment]
Thanks @Prafulkapse for sharing the details, the failure is coming during the notification failure (NOF) message generation, looks like the service content passed as an input to the "Wait For Response" action seems to be invalid which results in the failure. I will work on improving the error message in the upcoming release. Can you please confirm whether we expect the NOF generation here? I mean the reasons why the response or signal are not received in this case.  Please make sure the scenario are wired correctly as per https://techcommunity.microsoft.com/t5/integrations-on-azure-blog/building-end-to-end-rosettanet-application-using-logic-app/ba-p/2854346 on both Responder and Initiator side.

**Comment by praveensri** [comment]
@Prafulkapse , Please check with the customer , if the host and guest partner contact details (name, fax, telephone number etc) are updated. These are mandatory fields required by RosettaNet.

**Comment by Prafulkapse** [comment]
@praveensri  Yes, the host and guest partner contact details are mandatory. I am able to reproduce the error by removing them. But can the wait for response give a proper error message instead of a generic one ?

**Comment by praveensri** [comment]
Thanks @Prafulkapse for the confirmation. Yes I am working on improving the error response in this case, will share the ETA soon. So I am assuming the issue gets resolved if you provide the partner contact details, right?
